source/sas.cpp

```diff
@@ -756,40 +758,89 @@ SAS::Timestamp SAS::Marker::get_timestamp() const
 }
 
 
-void SAS::log_to_stdout(log_level_t level,
-                        const char *module,
-                        int line_number,
-                        const char *fmt,
-                        ...)
+void SAS::sasclient_log_callback(log_level_t level,
+                                const char *module,
+                                int line_number,
+                                const char *fmt,
+                                ...)
 {
   va_list args;
-  const char* level_str;
-
   va_start(args, fmt);
+  _sasclient_log_callback(level, module, line_number, fmt, args);
+  va_end(args);
+}
 
-  switch (level) {
-    case LOG_LEVEL_ERROR:   level_str = "ERROR"; break;
-    case LOG_LEVEL_WARNING: level_str = "WARNING"; break;
-    case LOG_LEVEL_STATUS:  level_str = "STATUS"; break;
-    case LOG_LEVEL_INFO:    level_str = "INFO"; break;
-    case LOG_LEVEL_VERBOSE: level_str = "VERBOSE"; break;
-    case LOG_LEVEL_DEBUG:   level_str = "DEBUG"; break;
-    default:                level_str = "UNKNOWN"; break;
+
+// Write the internally raised log to a char array, then pass
+// this to the common logging callback to log.
+void SAS::_sasclient_log_callback(log_level_t level,
+                                  const char *module,
+                                  int line_number,
+                                  const char *fmt,
+                                  va_list args)
+{
+  const char* log_level[] = {"Error", "Warning", "Status", "Info", "Verbose", "Debug"};
+  char logline[MAX_LOGLINE];
+  int written = 0;
+
+  // Strip the directory from the file location.
+  const char* mod = strrchr(module, '/');
+  module = (mod != NULL) ? mod + 1 : module;
+
+  if (line_number)
+  {
+    written = snprintf(logline, MAX_LOGLINE - 2, "CHECK %s %s:%d: ", log_level[level], module, line_number);
```

Comment:
Why is there this random CHECK in here?

Reply:
The log level gets passed through to the user on the log callback, so you don't need to render it to text here.